GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m

```diff
@@ -91,4 +105,40 @@ - (void)testLegacyCoding {
 }
 #endif // TARGET_OS_IOS || TARGET_OS_MACCATALYST
 
+- (void)testUpdateAuthState {
+  NSString *idToken = [self idTokenWithExpireTime:kIDTokenExpireTime];
+  OIDAuthState *authState = [OIDAuthState testInstanceWithIDToken:idToken
+                                                      accessToken:kAccessToken
+                                            accessTokenExpireTime:kAccessTokenExpireTime];
+  
+  GIDGoogleUser *user = [[GIDGoogleUser alloc] initWithAuthState:authState profileData:nil];
+  
+  XCTAssertEqualObjects(user.accessToken.tokenString, kAccessToken);
+  XCTAssertEqualWithAccuracy([user.accessToken.expirationDate timeIntervalSinceReferenceDate],
+                             kAccessTokenExpireTime, kTimeAccuracy);
+  XCTAssertEqualObjects(user.idToken.tokenString, idToken);
+  XCTAssertEqualWithAccuracy([user.idToken.expirationDate timeIntervalSinceReferenceDate],
+                             kIDTokenExpireTime, kTimeAccuracy);
+  
+  NSString *idTokenNew = [self idTokenWithExpireTime:kNewIDTokenExpireTime];
+  OIDAuthState *newAuthState = [OIDAuthState testInstanceWithIDToken:idTokenNew
+                                                         accessToken:kNewAccessToken
+                                               accessTokenExpireTime:kNewAccessTokenExpireTime];
+  
+  [user updateAuthState:newAuthState profileData:nil];
```

Comment:
How about testing the case where `profileData` is provided?

Reply:
Done.